src/bandwidth.rs

```diff
@@ -252,3 +223,13 @@ impl<TInner: AsyncWrite> AsyncWrite for BandwidthConnecLogging<TInner> {
         this.inner.poll_close(cx)
     }
 }
+
+/// Wraps around a `Future` that produces a connection. Wraps the connection around a bandwidth
+/// counter.
+#[deprecated(since = "0.50.1")]
+#[pin_project::pin_project]
+pub struct BandwidthFuture<TInner> {
```

Comment:
Changing `BandwidthLogging` to implement `StreamMuxer` instead of `Transport` is a breaking change. Given that you are proposing a breaking change anyways, why deprecate here?

Reply:
Oh, right. Will remove